internal/resources/kubernetessecret/data_source_secret.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package kubernetessecret
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	secretclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster"
+	secretexportclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/kubernetessecret/cluster/secretexport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func DataSourceSecret() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSecretRead,
+		Schema:      secretSchema,
+	}
+}
+
+func dataSourceSecretRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	config := m.(authctx.TanzuContext)
+
+	var (
+		diags            diag.Diagnostics
+		secretResp       *secretclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceGetSecretResponse
+		secretexportResp *secretexportclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceGetSecretExportResponse
+		err              error
+	)
+
+	secretName, _ := d.Get(NameKey).(string)
+
+	secretResp, err = config.TMCConnection.SecretResourceService.SecretResourceServiceGet(constructFullname(d))
+	if err != nil || secretResp == nil {
+		if clienterrors.IsNotFoundError(err) {
+			_ = schema.RemoveFromState(d, m)
+			return diags
+		}
+
+		return diag.FromErr(errors.Wrapf(err, "unable to get Tanzu Mission Control secret entry, name : %s", secretName))
+	}
+
+	status := map[string]interface{}{
+		"phase": secretResp.Secret.Status.Conditions["Ready"],
+	}
+
+	if d.Get(ExportKey).(bool) {
+		secretexportResp, err = config.TMCConnection.SecretExportResourceService.SecretExportResourceServiceGet(constructFullnameSecetExport(d))
+		if err != nil || secretexportResp == nil {
+			return diag.FromErr(errors.Wrapf(err, "unable to get Tanzu Mission Control secret export entry, name : %s", secretName))
+		}
+	}
+
+	status["export_phase"] = secretexportResp.SecretExport.Status.Conditions["Ready"]
```

Comment:
Should we have `export_phase` assigned to a key in constants?

Reply:
updated